tests/Unit/Domain/Test_AreaElement.cpp

```diff
@@ -157,11 +154,81 @@ void test_sphere_integral() {
           custom_approx(4. / 3. * M_PI * inner_radius * inner_radius));
   }
 }
-}  // namespace
+
+// checks that the integral over the horizon of a Kerr black hole
+// corresponds to the expected area
+void test_kerr_area() {
+  for (const auto& [ref_level, mass, spin] : cartesian_product(
+           std::array<size_t, 3>{{0, 1}}, make_array(0.001, 1., 100.),
+           std::array<std::array<double, 3>, 3>{{make_array(0., 0., 0.),
+                                                 make_array(0.1, 0.2, 0.3),
+                                                 make_array(0., 0., 0.99)}})) {
+    const double horizon_radius = mass * (1 + sqrt(1 - dot(spin, spin)));
+    domain::creators::Shell shell{
+        horizon_radius, 10. * horizon_radius, ref_level, {10, 10}, true};
+    const auto shell_domain = shell.create_domain();
+    const auto& blocks = shell_domain.blocks();
+    const auto& initial_ref_levels = shell.initial_refinement_levels();
+    const auto element_ids = initial_element_ids(initial_ref_levels);
+    const auto excision_sphere =
+        shell_domain.excision_spheres().at("ExcisionSphere");
+    const Mesh<2> face_mesh{10, Spectral::Basis::Legendre,
+                            Spectral::Quadrature::GaussLobatto};
+    domain::CoordinateMaps::KerrHorizonConforming kerr_schild_map(mass, spin);
+
+    // Shell does not distort to conform to the horizon in Kerr-Schild
+    // coordinates for spinning black holes, so we manually append the horizon
+    // conforming map
+    const auto grid_to_kerr_conforming_map =
+        CoordinateMap<Frame::Inertial, Frame::Inertial,
+                      domain::CoordinateMaps::KerrHorizonConforming>(
+            kerr_schild_map);
+    gr::Solutions::KerrSchild kerr_schild{mass, spin, {0., 0., 0.}};
+    double horizon_area = 0.;
+    for (const auto& element_id : element_ids) {
+      const auto& current_block = blocks.at(element_id.block_id());
+      const auto element_abutting_direction =
+          excision_sphere.abutting_direction(element_id);
+      if (element_abutting_direction.has_value()) {
+        const auto face_logical_coords = interface_logical_coordinates(
+            face_mesh, element_abutting_direction.value());
+        const ElementMap logical_to_grid_map(
```

Comment:
You can also use `CoordinateMaps::Composition` to get the combined logical -> grid -> kerr deformed map. Then you don't need to concatenate Jacobians yourself etc. You can get the ElementLogical -> BlockLogical map from `ElementToBlockLogicalMap.hpp`, then compose with `block.stationary_map()` and the Kerr conforming map.

Reply:
I am not sure how this is supposed to work. The problem is that the Kerr conforming map effectively goes from the intertial to the inertial frame, so I cannot put it directly into `Composition` because that will not be instantiated. 

So I have to somehow concatenate `block.stationary_map()` and the Kerr conforming map into a single `CoordinateMap` before, presumably using `make_coordinate_map`. But `block.stationary_map()` is of type `CoordinateMapBase` but `make_coordinate_map` needs the dynamic type at compile time as far as I can tell, so Im not sure how to do this.

Reply:
Here's an example:

```cpp
        domain::CoordinateMaps::Composition coord_map{
            domain::element_to_block_logical_map(element_id),
            current_block.stationary_map().get_to_grid_frame(),
            std::make_unique<
                CoordinateMap<Frame::Grid, Frame::Inertial,
                              domain::CoordinateMaps::KerrHorizonConforming>>(
                kerr_schild_map)};
        const auto inertial_coords = coord_map(face_logical_coords);
        const auto inv_jacobian = coord_map.inv_jacobian(face_logical_coords);
```

I tested that this compiles.

Reply:
I see, so we get the map to the grid frame so all the frames are different. Do you think it would be a good idea to put the entire composition in the header file so it is possible to also have maps that map to the same frame in there? Currently this is not possible because the instantiation would not be there.

This also threw up a bug #4851 which needs to merged before this test can pass.